crates/rpc/src/error.rs

```diff
@@ -1,14 +1,18 @@
 use beerus_core::CoreError;
 use eyre::Report;
+use jsonrpsee::core::Error;
 use jsonrpsee::types::ErrorObjectOwned;
 use starknet::core::types::StarknetError;
 use starknet::providers::ProviderError;
 use starknet::providers::ProviderError::StarknetError as StarknetProviderError;
+use thiserror::Error;
 
-#[derive(Debug)]
+#[derive(Error, Debug)]
 pub enum BeerusRpcError {
+    #[error(transparent)]
     Provider(ProviderError),
-    Other((i32, String)),
+    #[error("Other: `{code}` :: `{msg}`")]
+    Other { code: i32, msg: String },
```

Comment:
This is a breaking change to the public API. Is it necessary?

Reply:
To use thiserror, had to declare error message, and with current Other((i32, String)), I was not able to do that, because values are nested.
Alternative was to either separately implement Display trait for Other((i32, String)) or use a version like Other(i32, String) or what I implemented currently so values can be accessed

Reply:
Spec mismatch seems to be specific case of an error, so it doesn't have to match `Other`. Particularly `code` and `message` (supposedly for JSON-RPC response) do not make sense when specs mismatch, as we can't even have a server running in such case. I'd add an enum variant to represent the spec mismatch situation, with respective error message derived by `#[error("...")]`. Then there is no need to change `Other` variant.

Reply:
This looks go to me now. We could split the error type to have a specific one for the `run` method, avoiding the occurrence of this variant for regular RPC calls, but that can be done outside of this PR.

Thoughts @sergey-melnychuk ?